include/libdnf/base/goal.hpp

```diff
@@ -34,6 +34,15 @@ namespace libdnf {
 
 class Goal {
 public:
+    struct UsedDifferentSack : public RuntimeError {
```

Comment:
Is it `RuntimeError`? I agree with your opinion, you said during our call, that it is a serious programming error we don't want to catch. Please inherit from `LogicError`.